support new platform api, thermal and sfp part

stephenxs opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->


**- A picture of a cute animal (not mandatory but encouraged)**

keboliu commented on platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py:
```diff
@@ -95,5 +131,169 @@ def _get_port_position_tuple_by_sku_name(self):
+        /var/run/hwmanagement/system (which is defined as REBOOT_CAUSE_ROOT)
+        If a reboot cause file doesn't exists, returns '0'.
+        '''
+        return self._read_generic_file(join(REBOOT_CAUSE_ROOT, filename), REBOOT_CAUSE_FILE_LENGTH).rstrip('\n')
```
return true/false

stephenxs replied:
fixed

keboliu commented on platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py:
```diff
@@ -28,6 +30,37 @@
+REBOOT_CAUSE_ROOT = HWMGMT_SYSTEM_ROOT
+
+REBOOT_CAUSE_POWER_LOSS_FILE = 'reset_main_pwr_fail'
+REBOOT_CAUSE_THERMAL_OVERLOAD_CPU_FILE = ''
```
remove if we don't support?

stephenxs replied:
fixed

keboliu commented on platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py:
```diff
@@ -28,6 +30,37 @@
+REBOOT_CAUSE_POWER_LOSS_FILE = 'reset_main_pwr_fail'
+REBOOT_CAUSE_THERMAL_OVERLOAD_CPU_FILE = ''
+REBOOT_CAUSE_THERMAL_OVERLOAD_ASIC_FILE = 'reset_asic_thermal'
+REBOOT_CAUSE_THERMAL_OVERLOAD_OTHER_FILE = ''
```
remove if we don't support

stephenxs replied:
fixed

keboliu commented on platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py:
```diff
@@ -95,5 +129,169 @@ def _get_port_position_tuple_by_sku_name(self):
+        except:
+            return ''
+
+    def _read_reboot_cause_file(self, filename):
```
_verify_reboot_cause_by_reading_file()  ?

keboliu commented on platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py:
```diff
@@ -95,5 +129,169 @@ def _get_port_position_tuple_by_sku_name(self):
+        /var/run/hwmanagement/system (which is defined as REBOOT_CAUSE_ROOT)
+        If a reboot cause file doesn't exists, returns '0'.
+        '''
+        return REBOOT_DUE_TO_THIS_FILE == self._read_generic_file(join(REBOOT_CAUSE_ROOT, filename), REBOOT_CAUSE_FILE_LENGTH).rstrip('\n')
```
how about change to:  return bool(int(self._read_generic_file(join(REBOOT_CAUSE_ROOT, filename), REBOOT_CAUSE_FILE_LENGTH).rstrip('\n'))) ?  by this can remove 'REBOOT_DUE_TO_THIS_FILE '

Commit:
support new platform api, thermal and sfp part

keboliu commented on platform/mellanox/mlnx-platform-api/sonic_platform/psu.py:
```diff
@@ -26,30 +31,53 @@ def __init__(self, psu_index):
+            with open(filename, 'r') as fileobj:
+                result = int(fileobj.read())
+        except:
+            logger.log_info("Fail to read file {}, maybe it doesn't exist".format(filename))
```
suggest rewording it, "file {} doesn't exist."

keboliu commented on platform/mellanox/mlnx-platform-api/sonic_platform/psu.py:
```diff
@@ -60,15 +88,48 @@ def get_presence(self):
+            voltage = self._read_generic_file(os.path.join(self.psu_path, self.psu_voltage), 0)
+            return float(voltage) / 1000
+        else:
+            raise NotImplementedError
```
I am not sure whether we should raise "NotImplemenetedError" here, considering one case, if there is some bug with sysFS and the symbolic link was deleted, in this case maybe return a "none" would be better? just a thought.

stephenxs replied:
it seems your suggestion is better. will fix it.

keboliu commented on platform/mellanox/mlnx-platform-api/sonic_platform/thermal.py:
```diff
@@ -0,0 +1,216 @@
+        THERMAL_DEV_CATEGORY_PSU:(1, 2),
+        THERMAL_DEV_CATEGORY_CPU_PACK:(0,1)
+    },
+]
```
Add 3800?

Commit:
1. support ambient sensors for port, fan, combex, asic and board on varient SKUs
2. adjust name conventions of thermal sensors so that they are better aligned with what is displayed on user interface provided by other program
